src/renderers/WebGLRenderer.js

```diff
@@ -1334,6 +1334,8 @@ function WebGLRenderer( parameters = {} ) {
 		object.modelViewMatrix.multiplyMatrices( camera.matrixWorldInverse, object.matrixWorld );
 		object.normalMatrix.getNormalMatrix( object.modelViewMatrix );
 
+		material.onUpdate( _this, scene, camera, geometry, object, group );
```

Comment:
What do you think about naming it `material.onBeforeRender()` insted?
Just like the `object.onBeforeRender()` some lines up.

Reply:
For me okay, I think that `material.onBeforeRender()` is more familiar.

Reply:
It will be replace by `prototype.onBeforeRender` in `WebGLNodes`  and not documented equivalent to `Material.onBuild()`, without problems in that regards?

Reply:
otherwise could we make it "official"? with documentation?

Reply:
I say this because I would like to make sure that this method we only use this for `NodeMaterial`

Reply:
Should we remove `Material.onBuild()`?

Reply:
Still yes. 

Material.onBuild() is used for build GLSL code. 

Material.onUpdate() is used for update uniform values that need update per frame, like CameraNode for example that use the current renderer camera values.

Reply:
Okay, sounds good!